be/src/storage/lake/general_tablet_writer.cpp

```diff
@@ -105,14 +106,23 @@ Status HorizontalGeneralTabletWriter::flush_segment_writer(SegmentPB* segment) {
 
 VerticalGeneralTabletWriter::VerticalGeneralTabletWriter(TabletManager* tablet_mgr, int64_t tablet_id,
                                                          std::shared_ptr<const TabletSchema> schema, int64_t txn_id,
-                                                         uint32_t max_rows_per_segment)
-        : TabletWriter(tablet_mgr, tablet_id, std::move(schema), txn_id), _max_rows_per_segment(max_rows_per_segment) {}
+                                                         uint32_t max_rows_per_segment, ThreadPool* flush_pool)
+        : TabletWriter(tablet_mgr, tablet_id, std::move(schema), txn_id, flush_pool),
+          _max_rows_per_segment(max_rows_per_segment) {}
 
 VerticalGeneralTabletWriter::~VerticalGeneralTabletWriter() = default;
 
 // To developers: Do NOT perform any I/O in this method, because this method may be invoked
 // in a bthread.
 Status VerticalGeneralTabletWriter::open() {
+    if (_flush_pool != nullptr) {
+        // Use CONCURRENT mode to ensure segments can finalize in parallel
+        _segment_writer_finalize_token =
+                std::make_unique<ConcurrencyLimitedThreadPoolToken>(_flush_pool, _flush_pool->max_threads() * 2);
+        if (_segment_writer_finalize_token == nullptr) {
+            return Status::InternalError("Segment writer's flush token init failed");
+        }
```

Comment:
```suggestion
```

Reply:
`std::make_unique` will never return a nullptr, so this check will always be false.